internal/pkg/cli/task_run.go

```diff
@@ -751,9 +763,16 @@ func (o *runTaskOpts) buildAndPushImage() error {
 		additionalTags = append(additionalTags, o.imageTag)
 	}
 
+	var dockerContext string
+	if o.dockerfileContextPath == "" {
+		dockerContext = filepath.Dir(o.dockerfilePath)
+	} else {
+		dockerContext = o.dockerfileContextPath
+	}
+
 	if _, err := o.repository.BuildAndPush(dockerengine.New(exec.NewCmd()), &dockerengine.BuildArguments{
 		Dockerfile: o.dockerfilePath,
-		Context:    filepath.Dir(o.dockerfilePath),
+		Context:    dockerContext,
```

Comment:
I wonder if we can set the default value for `o.dockerfileContextPath` in `newTaskRunOpts()` to have the default value to be
```go
if o.dockerfileContextPath == "" {
  o.dockerfileContextPath = filepath.Dir(o.dockerfilePath)
}
```
Then we can have `Context: o.dockerfileContextPath` here.

Reply:
I did this slightly differently but it's essentially the same if-statement